web-share/src/js/service-worker.js

```diff
@@ -0,0 +1,82 @@
+import {CacheOnly} from 'workbox-strategies';
+import {clientsClaim, skipWaiting} from 'workbox-core';
+import {precacheAndRoute} from 'workbox-precaching';
+import {RangeRequestsPlugin} from 'workbox-range-requests';
+import {registerRoute} from 'workbox-routing';
+
+import {cacheName, channelName, urlPrefix} from './constants';
+import {mimeRoute as audioRoute} from '../svelte/routes/Audio.svelte';
+import {mimeRoute as imagesRoute} from '../svelte/routes/Images.svelte';
+import {mimeRoute as videosRoute} from '../svelte/routes/Videos.svelte';
+
+const broadcastChannel = BroadcastChannel ? new BroadcastChannel(channelName) : null;
+
+const shareTargetHandler = async ({event}) => {
+  if (broadcastChannel) {
+    broadcastChannel.postMessage('Saving media locally...');
+  }
+
+  const formData = await event.request.formData();
+  const mediaFiles = formData.getAll('media');
+  const cache = await caches.open(cacheName);
+
+  for (const mediaFile of mediaFiles) {
+    // TODO: Instead of bailing, come up with a
+    // default name for each possible MIME type.
+    if (!mediaFile.name) {
+      if (broadcastChannel) {
+        broadcastChannel.postMessage('Sorry! No name found on incoming media.');
+      }
+      continue;
+    }
+    await cache.put(
```

Comment:
Why await here?
You could add every `cache.put` promise and wait with `Promise.all` at the end right?

Reply:
I didn't _put_ a huge amount of thought into it, but I guess the current approach has two advantages, given that each `mediaFile` might potentially be quite large:

- Caching one-by-one seems like it would play nicer if our PWA is close to using up their storage quota on a given device.

- I could imagine that streaming the `mediaFile` bodies into new `Response` objects in the course of writing them to cache storage could be memory intensive, and doing that one at a time instead of a bunch in parallel seems less likely to degrade performance.

But... like I said, it's not something that I really thought about ahead of time.

(Before `async`/`await`, when everything had to be chained together via `.then()`, I was more inclined to use things like `.map()` + `Promise.all()`, but I do that less now as a matter of style.)